tests/integration/kubernetes/k8s-confidential.bats

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bats
+# Copyright 2022-2023 Advanced Micro Devices, Inc.
+# Copyright 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+IMAGE_REPO="ghcr.io/confidential-containers/test-container"
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
+	# having substrings, like qemu, being matched with qemu-$something.
+	( [[ ! " ${SUPPORTED_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]] ) && skip "Test not supported yet."
```

Comment:
My understanding (which might be wrong) is that bats runs setup before every test and that a skip in setup will skip all the tests, so I'm not sure if we need this logic in the test run and teardown?

Reply:
A `skip` on `setup()` will skip all tests, so you don't need to put a skip in the tests individually. However, the `teardown()` still runs, if you want to skip it then put the `skip` there too.

A `skip` on tests will skip only individual tests, `setup()` and `teardown()` still run.

Reply:
Thanks @stevenhorsman and @wainersm for your feedback. I have changed my code so that we are skipping the test in `setup_file` so that all the tests will be skipped if it is not one of the supported hypervisors. And we can just check the `BATS_TEST_SKIPPED` variable in `teardown_file` to decide whether we should clean it up or not.